src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -435,33 +436,43 @@ public IReadOnlyList<LoggerClass> GetLogClasses(IEnumerable<ClassDeclarationSynt
                 return (loggerField, false);
             }
 
-            private (int eventId, int? level, string message, string? eventName) ExtractAttributeValues(AttributeArgumentListSyntax args, SemanticModel sm)
+            private (int eventId, int? level, string message, string? eventName, bool skipEnabledCheck) ExtractAttributeValues(AttributeArgumentListSyntax args, SemanticModel sm)
             {
-                int eventId = 0;
+                if (args == null)
+                {
+                    // e.g. is null as [LoggerMessage]
+                    return (-1, null, string.Empty, null, false);
+                }
+                int eventId = -1; // default eventId value when not provided is -1
                 int? level = null;
                 string? eventName = null;
                 string message = string.Empty;
+                bool skipEnabledCheck = false;
                 foreach (AttributeArgumentSyntax a in args.Arguments)
                 {
                     // argument syntax takes parameters. e.g. EventId = 0
                     Debug.Assert(a.NameEquals != null);
                     switch (a.NameEquals.Name.ToString())
                     {
                         case "EventId":
-                            eventId = (int)sm.GetConstantValue(a.Expression, _cancellationToken).Value!;
+                            eventId = (int?)sm.GetConstantValue(a.Expression, _cancellationToken).Value ?? -1;
                             break;
                         case "EventName":
                             eventName = sm.GetConstantValue(a.Expression, _cancellationToken).ToString();
                             break;
                         case "Level":
-                            level = (int)sm.GetConstantValue(a.Expression, _cancellationToken).Value!;
+                            level = (int?)sm.GetConstantValue(a.Expression, _cancellationToken).Value;
                             break;
                         case "Message":
                             message = sm.GetConstantValue(a.Expression, _cancellationToken).ToString();
                             break;
+                        case "SkipEnabledCheck":
+                            bool? flag = (bool?)sm.GetConstantValue(a.Expression, _cancellationToken).Value;
```

Comment:
What happens to this code if someone writes code that might not compile? Does the source generator fail with an exception, adding more errors to the Error List?

```C#
[LoggerMessage(SkipEnabledCheck = 6)]
```

Reply:
for this specific use case we get this error:

```
[warning CS8785: Generator 'LoggerMessageGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'InvalidCastException' with message 'Unable to cast object of type 'System.Int32' to type 'System.Nullable`1[System.Boolean]'.']
```

Reply:
Should we fix that? That may be confusing to users.

Reply:
OK looking into it.

Reply:
Fixed and made generator more robust against misconfigured input in latest commit

/cc @chsienki